### PR TITLE
Add support to history event generator to generate multi-branches of history events

### PR DESCRIPTION
**Pull request by yux0**
Add support to history event generator to generate multi-branches of history events

### CODE REVIEW

File: common/xdc/event_generator.go [hunk 1]
```diff
@@ -94,28 +98,29 @@ func NewEventGenerator() Generator {
 		randomEntryVertices: make([]Vertex, 0),
 		dice:                rand.New(rand.NewSource(time.Now().Unix())),
 		canDoBatch:          defaultBatchFunc,
-		resetPoints:         []ResetPoint{initialResetPoint},
+		resetPoints:         make([]ResetPoint, 0),
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: make([]ResetPoint, 0) ->  make([]ResetPoint)

File: common/xdc/event_generator.go [hunk 2]
```diff
@@ -148,37 +153,21 @@ func (g *EventGenerator) GetNextVertices() []Vertex {
...
-		previousVertices: previousVerticesSnapshot,
-		leafVertices:     leafVerticesSnapshot,
-	}
-	g.resetPoints = append(g.resetPoints, newResetPoint)
+	g.addNewResetPoint()
 	return batch
 }
 
+// ResetAsNew resets history event generator to its initial state
+func (g *EventGenerator) ResetAsNew() {
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: just Reset?

File: common/xdc/event_generator.go [hunk 3]
```diff
@@ -196,11 +186,54 @@ func (g *EventGenerator) ListResetPoint() []ResetPoint {
 	return g.resetPoints
 }
 
-// SetCanDoBatchOnNextVertex sets a function to determine next generated batch of history events
-func (g *EventGenerator) SetCanDoBatchOnNextVertex(canDoBatchFunc func([]Vertex) bool) {
+// RandomReset randomly pick a reset point and reset the event generator to the point
+func (g *EventGenerator) RandomReset() int {
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: RendomizedResetToPointInTime? something like that

Comment (wxing1292) [reply common/xdc/event_generator.go]: or RendomResetToResetPoint

Comment (yux0) [reply common/xdc/event_generator.go]: pick RendomResetToResetPoint

File: common/xdc/event_generator.go (hunk 2 above)
Comment (yux0) [reply common/xdc/event_generator.go]: updated

**Pull request merged**